src/btree/col_srch.c

```diff
@@ -286,11 +286,39 @@ __wt_col_search(
      * because column-store files are dense, but in this case the caller searched past the end of
      * the table.
      */
-    cbt->ins_head = WT_COL_APPEND(page);
-    if ((cbt->ins = __col_insert_search(cbt->ins_head, cbt->ins_stack, cbt->next_stack, recno)) ==
-      NULL)
+    ins_head = WT_COL_APPEND(page);
+    ins = __col_insert_search(ins_head, cbt->ins_stack, cbt->next_stack, recno);
```

Comment:
I believe it's correct as-is, since the only two cases where we fail to set `cbt->ins`/`cbt->ins_head` are the cases where the record number is out of bounds somehow (i.e. `cbt->recno == WT_RECNO_OOB` or `cbt->recno != WT_INSERT_RECNO(ins)`). I don't think other code should be looking at the append list at all in these cases, but I could be wrong. 

I guess we could hit the case where where `WT_SKIP_LAST` returned `NULL` for a valid record number, but I suspect the skip list would have to be messed up already for that to happen. Perhaps a `WT_ASSERT` that `WT_SKIP_LAST` is non-null would be appropriate.

That said, I'm happy enough to add some catch-all NULL assignments out of paranoia, but I think that could hide a logic error in testing.

Reply:
> That said, I'm happy enough to add some catch-all NULL assignments out of paranoia, but I think that could hide a logic error in testing.

That's my preference. I think it would be much harder to debug/diagnose a bug where those fields have random outdated data, than one where the fields are NULL.

Reply:
Yeah, that's a fair point. I've added this in 3e929de.